sklearn/feature_extraction/dict_vectorizer.py

```diff
@@ -94,11 +95,34 @@ class DictVectorizer(BaseEstimator, TransformerMixin):
     """
 
     def __init__(self, dtype=np.float64, separator="=", sparse=True,
-                 sort=True):
+                 sort=True, n_features=None):
+        if n_features is not None:
+            self._validate_params(n_features)
+
         self.dtype = dtype
         self.separator = separator
         self.sparse = sparse
         self.sort = sort
+        self.n_features = n_features
+
+    @staticmethod
+    def _validate_params(n_features):
+        # strangely, np.int16 instances are not instances of Integral,
+        # while np.int64 instances are...
+        if not isinstance(n_features, (numbers.Integral, np.integer)):
+            raise TypeError("n_features must be integral, got %r (%s)."
+                            % (n_features, type(n_features)))
+        elif n_features < 1 or n_features >= 2 ** 31:
+            raise ValueError("Invalid number of features (%d)." % n_features)
+    
+    @staticmethod
+    def _validate_feature_name(feature, x):
+        # strangely, np.int16 instances are not instances of Integral,
```

Comment:
I don't see what this has to do with it...?

Reply:
Hahah yeah, I not sure about this too!
Right now code creates dummy features as integers and if an input feature has a name where `feature_name.isdigit() == True` it throws an error.

Now I'm not sure if I should indeed create a `dict` with all the dummy features initially or if it's better just to add the dummy values latter. 

But as you said I think this will become more clear when I start with documentation and tests, I'll do that :)

Thank you very much for your comments, they were very helpful :)

Reply:
Keep the fake features to `get_feature_names`. They shouldn't actually "exist" in the `vocabulary_`.